WTWSMS/decisions/WtWSMS_diploaction_decisions.txt

```diff
@@ -96,4 +96,84 @@ targetted_decisions = {
 			factor = 1
 		}
 	}
+	
+	# TODO: ask_to_become_foederati_emperor  (Empire->barbarians)
+	
+	become_foederati_barbarian = { # barbarians->Empire
+		only_independent = yes
+		ai_check_interval = 12
+
+		potential = {
+			OR = {
+				has_landed_title = e_byzantium
+				has_landed_title = e_wre
+				has_landed_title = e_roman_empire
+			}
+		}
+	
+		from_potential = {
+			independent = yes
+			mercenary = no
+			holy_order = no
+			is_landed = yes
+		}
+
+		allow = {
+			FROM = {
+				show_scope_change = no
+				war = no
+				OR = {
+					custom_tooltip = { #Borders the Eastern Roman Empire
+						text = become_foederati_barbarian_allow_borders_byzantium_tt
+						any_neighbor_independent_ruler = {
+							has_landed_title = e_byzantium
+						}
+					}
+					custom_tooltip = { #Borders the Western Roman Empire
+						text = become_foederati_barbarian_allow_borders_wre_tt
+						any_neighbor_independent_ruler = {
+							has_landed_title = e_wre
+						}
+					}
+					custom_tooltip = { #Borders the Roman Empire
+						text = become_foederati_barbarian_allow_borders_roman_empire_tt
+						any_neighbor_independent_ruler = {
+							has_landed_title = e_roman_empire
+						}
+					}
+				}
+				NOT = {
+					is_tributary = yes
+				}
+				NOT = { tier = emperor }
+				custom_tooltip = {
+					text = become_foederati_barbarian_cooldown_tt
+					NOT = { has_character_modifier = become_foederati_timer }
+				}
+				prisoner = no
+				NOT = { trait = incapable }
+				is_inaccessible_trigger = no
+			}
+		}
+		
+		effect = {
+			make_tributary = {
+				who = FROM
+				tributary_type = foederati
+			}
+			custom_tooltip = { text = chinese_grace_become_tributary_effect_TT }
+			FROM = {
+				sound_effect = china_grace_gain
+			}
```

Comment:
Why does this connect to China?

Reply:
I forgot to replace it with the name of a dedicated tooltip.

Reply:
Done.